Add support for `delete` operator on UDVT

xBA5ED opened the pull request:
Fixes #12273

xBA5ED committed:
Add support for delete operator on UDVT

xBA5ED committed:
Add support for delete operator on UDVT

ekpyron commented:
This would need semantics tests checking for sane behavior and a changelog entry, if we're to merge it.

xBA5ED committed:
test: add semantic test for UDVT delete operator

xBA5ED committed:
test: add semantic test for UDVT delete operator

xBA5ED committed:
Merge branch 'support-udvt-delete' of github.com:xBA5ED/solidity into support-udvt-delete

xBA5ED committed:
docs: add UDVT delete operator to the changelog

xBA5ED commented:
I added semantic tests, the changelog entry and rebased the PR. If there is missing test coverage or if there are any other changes needed let me know.
